### PR TITLE
BaSP-TG-78: Super Admin

### PR DESCRIPTION
**Pull request by CamilaFigue**
- Super Admin table list
- Delete with edit icon
- Delete feature
- Redirect to add form
- Warning message for delete
- Add route on layout component
- Create form
- Add feature
- Info previous from form edition
- Edit feature
- Succes message on add or edit

### CODE REVIEW

File: src/Components/SuperAdmins/Modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// import React from 'react';
```
Comment (cocolobos) [inline src/Components/SuperAdmins/Modal/Modal.jsx:1]: Remove commented code

File: src/Components/Projects/List.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Comment (marcomigliaro) [inline src/Components/Projects/List.jsx:1]: Could remove this file

Comment (marcomigliaro) [inline src/Components/Projects/List.jsx:1]: Could remove this file

File: src/Components/SuperAdmins/ListItem.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        <td>{item.name}</td>
+        <td>{item.lastName}</td>
+        <td>{item.email}</td>
+        <td>{item.password}</td>
+        <td>{item.active.toString()}</td>
+        <td>
+          <button onClick={() => editSuperAdmin(item._id)}>Edit</button>
+        </td>
+        <td>
+          <button onClick={() => handleDelete(item._id)}>X</button>
```
Comment (marcomigliaro) [inline src/Components/SuperAdmins/ListItem.jsx:22]: Could do directly the deleteitem function here, and not use the handleDelete

File: src/Components/SuperAdmins/Modal/modal.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/* .modal {
```
Comment (marcomigliaro) [inline src/Components/SuperAdmins/Modal/modal.css:1]: Remove commented code

Comment (marcomigliaro) [inline src/Components/SuperAdmins/Modal/modal.css:1]: Remove commented code

File: src/Components/SuperAdmins/SuperAdminForm.jsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            name="lastName"
+            value={lastName}
+            onChange={(event) => setLastName(event.target.value)}
+          ></input>
+        </div>
+        <div>
+          <label>Email</label>
+          <input
+            type="text"
+            name="email"
```
Comment (marcomigliaro) [inline src/Components/SuperAdmins/SuperAdminForm.jsx:61]: Could be type email

Comment (marcomigliaro) [inline src/Components/SuperAdmins/SuperAdminForm.jsx:61]: Could be type email

File: src/Components/SuperAdmins/SuperAdminFormEdit.jsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            name="lastName"
+            value={lastName}
+            onChange={(event) => setLastName(event.target.value)}
+          ></input>
+        </div>
+        <div>
+          <label>Email</label>
+          <input
+            type="text"
+            name="email"
```
Comment (marcomigliaro) [inline src/Components/SuperAdmins/SuperAdminFormEdit.jsx:69]: Type could be email

**Review by marcomigliaro — APPROVED** [review]

File: src/Components/Projects/List.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+// import ListItem from './ListItem.jsx';
+import './list.module.css';
+
+const List = ({ list }) => {
+  console.log('Probando 2', list);
```
Comment (cocolobos) [inline src/Components/Projects/List.jsx:6]: remove console log

File: src/Components/SuperAdmins/Modal/Modal.jsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+// import Modal from './modal.css';
```
Comment (cocolobos) [inline src/Components/SuperAdmins/Modal/Modal.jsx:2]: do we need this code commented ?

File: src/Components/SuperAdmins/Modal/modal.css [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    background: #fdc800;
+    font-size: 18px;
+    font-weight: bold;
+    border-radius: 4px;
+    cursor: pointer;
+}
+
+.cancel {
+    background-color: gray;
+} */
```
Comment (cocolobos) [inline src/Components/SuperAdmins/Modal/modal.css:26]: do we need this code commented ?

File: src/Components/SuperAdmins/SuperAdminForm.jsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+        <div>
+          <label>Active</label>
+          <input
+            type="text"
+            name="active"
+            value={active}
+            onChange={(event) => setActive(event.target.value)}
+          ></input>
+        </div>
+        <button type="submit" onClick={() => alertSuccessfully()}>
```
Comment (cocolobos) [inline src/Components/SuperAdmins/SuperAdminForm.jsx:87]: maybe after submit on adding or editing the best is to redirect to the table component

File: src/Components/SuperAdmins/SuperAdminFormEdit.jsx [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+  const [active, setActive] = useState();
+
+  const params = window.location.search;
+  let id = params.substring(2);
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+
+    const url = `https://coco-trackgenix-server.vercel.app/superAdmins/${id}`;
+    console.log(id);
```
Comment (cocolobos) [inline src/Components/SuperAdmins/SuperAdminFormEdit.jsx:26]: remove console log

File: src/Components/SuperAdmins/index.js [hunk 12]
```diff
@@ -1,11 +1,58 @@
...
+        <h2>Super Admin</h2>
+        <SuperAdminForm switcher={switcher} />
+        <button onClick={switcher}>Back</button>
+      </section>
+    );
+  } else {
+    return (
+      <section className={styles.container}>
+        <h2>Super Admin</h2>
+        {/* <button onClick={() => editSuperAdmin(list._id)}>Edit</button> */}
```
Comment (cocolobos) [inline src/Components/SuperAdmins/index.js:49]: remove commented code

File: src/Components/SuperAdmins/Modal/Modal.jsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
```
Comment (cocolobos) [inline src/Components/SuperAdmins/Modal/Modal.jsx:3]: can we remove the commented code ?

**Pull request merged**